experimental/checkpoint_restore.md

```diff
@@ -0,0 +1,154 @@
+# Docker Checkpoint & Restore
+
+Checkpoint & Restore is a new feature that allows you to freeze a running
+container by checkpointing it, which turns its state into a collection of files
+on disk. Later, the container can be restored from the point it was frozen.
+
+This is accomplished using a tool called [CRIU](http://criu.org), which is an
+external dependency of this feature. A good overview of the history of
+checkpoint and restore in Docker is available in this
+[Kubernetes blog post](http://blog.kubernetes.io/2015/07/how-did-quake-demo-from-dockercon-work.html).
+
```

Comment:
Those limitations would seem to only matter in the --force case, and there's already a comment there noting that it requires you to configure the same base container. I'll add a sentence explicitly noting that a docker commit may be necessary.

Should we, for now, completely disallow checkpointing non detached containers?